LLVM_FOR_RISCV.md

```diff
@@ -0,0 +1,123 @@
+
+# Building LLVM on Linux for RISC-V 64-bit Cross-Compilation
+
+This document outlines the steps to build LLVM on a Linux machine for cross-compiling C/C++ code for RISC-V 64-bit.
+
+## Before You Start
+
+Update your package manager to avoid errors during setup:
+
+```bash
+sudo apt-get update
+# or
+sudo apt update
+```
+
+## Setup RISC-V and LLVM Environments
+
+Install the required packages:
+
+```bash
+sudo apt-get -y install \
+  binutils build-essential libtool texinfo \
+  gzip zip unzip patchutils curl git \
+  make cmake ninja-build automake bison flex gperf \
+  grep sed gawk python3 bc \
+  zlib1g-dev libexpat1-dev libmpc-dev \
+  libglib2.0-dev libfdt-dev libpixman-1-dev
+```
+
+## Directory Setup for RISC-V and LLVM Files
+
+Create a directory for RISC-V files:
+
+```bash
+cd /data/users/$USER/condor # or your preferred workspace
+mkdir llvm
+cd llvm
+export PATH=/data/tools/bin:$PATH # adjust to your chosen install path
+hash -r  # cleans the command hash to ensure the environment is up to date
+```
+
+## Cloning and Building the RISC-V GNU Toolchain
+
+Ensure the repository link is live before cloning:
+
+```bash
+git clone --recursive https://github.com/riscv/riscv-gnu-toolchain
+pushd riscv-gnu-toolchain
+./configure --prefix=/data/tools --enable-multilib # adjust to your chosen install path
+make -j 32
+```
+
+*Optionally, build QEMU for emulation. QEMU is a generic and open source machine emulator and virtualizer. It can be useful for emulating the RISC-V architecture, allowing you to test your compiled programs*
+
+```bash
+make -j 32 build-qemu
+```
+
+Then, return to your directory for RISC-V and LLVM files
+
+```bash
+popd
+```
+
+## Cloning and Building LLVM for RISC-V
+
+Clone the LLVM project and initiate the build (Ensure the repository link is live before cloning):
+
```

Comment:
what does this mean? how would I ensure the repo link is 'live'? and why do i need to?